### DISCUSSION TITLE
How to couple 2D heat conduction equations with 1D fluid conservation equations?

### DISCUSSION THREAD

**Discussion by Raven-pro** [idaholab/moose#22905] [Q&A General] [answered]
Hello everyone!

I have developed a single-phase flow application based on three conservation equations. And I want to couple solid heat conduction equations with it to achieve conjugate heat transfer.

As the flow model is 1D, for each node of the channel in, like z direction, there should be a solid heat conduction equation to transfer heat in the x direction.  

So this is a problem of coupling a 2D heat conduction equation and 1D fluid conservation equations, as I should use the temperature of the boundary of the heat structure to calculate heat transfer from solid to fluid.

How can I achieve it? THX!

**Comment by GiudGiud** [comment, marked as answer]
Hello

You will want to use heat structures for now, see the tutorial
https://mooseframework.inl.gov/docs/site/modules/thermal_hydraulics/tutorials/single_phase_flow/step02.html#step-2-conjugate-heat-transfer

In a few months we ll be able to run other modules (such a multi-D heat conduction using the heat conduction module) in combination with THM, we ll keep you posted. 

Guillaume

**Comment by friedmud** [comment]
You can put something together yourself pretty easily.

Use [GapValueAux](https://mooseframework.inl.gov/source/auxkernels/GapValueAux.html) in each direction to get the temperature on the other side of the gap in each direction... then simply apply [CoupledConvectiveHeatFlux](https://mooseframework.inl.gov/source/bcs/CoupledConvectiveHeatFluxBC.html) on both sides.

Now - this won't be perfectly conservative... but with a reasonable mesh it will work well enough (it's what we used for many years before we got the more sophisticated schemes in [GapHeatTransfer](https://mooseframework.inl.gov/source/bcs/GapHeatTransfer.html)
